crates/holochain_sqlite/src/sql/cell/schema/6-up.sql

```diff
@@ -0,0 +1,95 @@
+CREATE TABLE Warrant (
+  hash BLOB PRIMARY KEY ON CONFLICT IGNORE,
+  author BLOB NOT NULL,
+  timestamp INTEGER NOT NULL,
+  warrantee BLOB NOT NULL,
+  type TEXT NOT NULL,
+  blob BLOB NOT NULL
+);
+
+-- Drop foreign key constraint on action_hash in DhtOp.
+-- To achieve that, create a new table without it and drop the old table.
+CREATE TABLE IF NOT EXISTS DhtOpNew (
+  hash BLOB PRIMARY KEY ON CONFLICT IGNORE,
+  type TEXT NOT NULL,
+  basis_hash BLOB NOT NULL,
+  require_receipt INTEGER NOT NULL,  -- BOOLEAN
+  -- This is not strictly an action hash, but a foreign key to a row in the Action or Warrant table.
+  action_hash BLOB NOT NULL,
```

Comment:
I know we had some discussion about this on mattermost, but now that I'm seeing the code I understand it more clearly.

Would it be clearer to:
- add another field `warrant_hash`
- Both `action_hash` and `warrant_hash` are nullable
- Both `action_hash` and `warrant_hash` are foreign keys to their respective tables

Or should i just appreciate this is a stepping stone refactor and let it be :smile:

Reply:
Maybe is clearer. I consider this a "fix the Action table problem" PR, not a "stabilize warrants" one. I'd be happiest if I could just put this to bed and come back to it when it's warranting time.

Reply:
Working on feature gated pieces of code has been highly cumbersome here.

Reply:
Seems like it would be relevant to the changes in this PR, but if its very cumbersome I'm fine with deferring.

Reply:
You had suggested earlier there should be two tables, one for chain ops and one for warrant ops. Maybe that's the better solution. But as long as we haven't even decided on whether we're going to include warrants in 1.0, I don't want to spend more time on this.